libs-scala/scala-utils/src/main/scala/com/daml/scalautil/NeverEqualsOverride.scala

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.scalautil
+
+/** This trait is useful to be mixed in if:
+  * - equality semantics for the base type is not an issue
+  * - class has non-comparable typed fields
+  *
+  * Practical usage: mixing in in DTO ADT top trait, which has case class implementors having Array fields: disables equality semantics, and prevents Wartremover warnings.
+  */
+trait NeverEqualsOverride {
+  override def equals(o: Any): Boolean = false
```

Comment:
It might be better to simulate `java.lang.Object`'s behavior here.

```suggestion
trait NeverEqualsOverride {
  override final def equals(o: Any): Boolean = o match {
    case ar: AnyRef => this eq ar
    case _ => false
  }
  override final def hashCode = System.identityHashCode(this)
```

Reply:
If you find that disagreeable, then alternatively

```suggestion
trait NeverEqualsOverride extends Equals {
  override final def canEqual(o: Any) = false
  override final def equals(o: Any): Boolean = false
```

Reply:
If you're really committed to this, you should make `equals` throw an exception. 👻